ops/testing.py

```diff
@@ -1429,6 +1436,53 @@ def trigger_secret_expiration(self, secret_id: str, revision: int, *,
             label = secret.label
         self.charm.on.secret_expired.emit(secret_id, label, revision)
 
+    def get_filesystem_root(self, container: Union[str, Container]) -> pathlib.Path:
+        """Return the temp directory path harness will use to simulate the container filesystem.
+
+        In a real container runtime, each container has an isolated root filesystem.
+        To simulate this behaviour, the testing harness manages a temporary directory for
+        each container. Any Pebble filesystem API calls will be translated
+        and mapped to this directory, as if the directory was the container's
+        filesystem root.
+
+        This process is quite similar to the ``chroot`` command. Charm tests should
+        treat the returned directory as the container's root directory (``/``).
+        The testing harness will not create any files or directories inside the
+        simulated container's root directory; it's up to the test to populate the container's
+        root directory with any files or directories the charm needs.
+
+        Regarding the file ownership: unprivileged users are unable to create files with distinct
+        ownership. To circumvent this limitation, the testing harness maps all user and group
+        options related to file operations to match the current user and group.
+
+        Example usage:
+
+        >>> # charm.py
+        >>> import ops
+        >>> class ExampleCharm(ops.CharmBase):
+        >>>     def __init__(self, *args):
+        >>>         super().__init__(*args)
+        >>>         self.hostname = open("/etc/hostname").read()
+
+        >>> # test_charm.py
+        >>> from ops.testing import Harness
+        >>> harness = Harness(ExampleCharm)
+        >>> root = harness.get_filesystem_root("mycontainer")
+        >>> (root / "etc" / "hostname").write_text("hostname.example.com")
+        >>> harness.begin()
```

Comment:
I'd prefer to avoid the `>>> ` prefix on each line, as this isn't typically something you'd type into the Python REPL. To do this, use a double-colon on `Example usage::` and then make the follow code block indented by four spaces:

```suggestion
        Example usage::

            # charm.py
            import ops
            class ExampleCharm(ops.CharmBase):
                def __init__(self, *args):
                    super().__init__(*args)
                    self.hostname = open("/etc/hostname").read()

            # test_charm.py
            from ops.testing import Harness
            harness = Harness(ExampleCharm)
            root = harness.get_filesystem_root("mycontainer")
            (root / "etc" / "hostname").write_text("hostname.example.com")
            harness.begin()
```